### DISCUSSION TITLE
Why are workflows often times hitting the GitHub Actions timeout (6h) when using additional countries?

### DISCUSSION THREAD

**Discussion by matthias-seifert** [microsoft/AL-Go#1341] [Q&A]
We have some apps for AppSource that we build against the localizations for US, AT and CH using the additionalCountries setting in the .AL-Go/settings.json.

The GitHub workflows (mostly the Test Next Major workflow) will sometimes fail after 6 hours due to the GitHub Actions default timeout setting (360m). However, if the run does not fail, it only takes around 50min to 1h to succeed.

The app is being compiled once, and additional containers are being created for every localization of BC.
The last executed step is often times the creation of an additional container, but sometimes another step like testing the app etc.
![pipeline-timeout-container-creation](https://github.com/user-attachments/assets/e60b0734-6240-42ba-bda1-1120c8187ab9)

After creating the second-to-last container, the GitHub output shows sufficcient free memory to create at least 2 more containers:
![containter-memory](https://github.com/user-attachments/assets/07b8c61b-8aae-4258-b878-7f562b81e918)

I have noticed the frequency of this issue appearing become less frequent over the last 2 months.
Two months ago:
![image](https://github.com/user-attachments/assets/a381735d-3257-4a1d-ad59-348fc498741d)

This month:
![image](https://github.com/user-attachments/assets/5e7b4f23-ce03-44b8-8334-60a538928d20)

Still, this is billable time for our GitHub subscription. Any ideas what could be causing this behaviour?

**Comment by aholstrup1** [comment]
@matthias-seifert Not really a fix/answer to what you're asking for, but AL-Go should probably have an option to define timeouts. If you were able to time out your workflows after 1,5 hours, it would save some of the wasted compute time in the event that something stalls. 

I added an item to our backlog for this:
https://github.com/orgs/microsoft/projects/521/views/1?pane=issue&itemId=89142230

**Comment by freddydk** [comment]
@matthias-seifert - could you include or email the full log of the tests, which are failing, to see where exactly it is hanging.

**Reply by matthias-seifert** [reply]
Thanks - just sent you an email including some logs.

**Comment by freddydk** [comment]
@matthias-seifert - could you try to add this setting to your .github/AL-Go-Settings.json file:

"killPsSessionProcess": true

and then retry the workflows.
